src/components/EachMovie.js

```diff
@@ -1,6 +1,25 @@
 import { useParams } from "react-router";
-export const EachMovie = () =>{
-    console.log('Movie Description')
-    const name = useParams().name;
-    console.log(name)
+import { useMoviesData } from "../hooks/useMoviesData";
+import { MovieDescription } from "./MovieDescription";
+
+export const EachMovie = () => {
+    const mapping = { 'thriller': 0, 'romance': 1, 'horror': 2, 'action': 3, 'comedy': 4 }
+    const thing = useParams().id;
+    const arr = thing.split('-')
+    const id = arr[arr.length - 1]
+    const tmp = useParams().category;
+    const cat = tmp === 'movies' ? arr[arr.length - 2] + '-Movies' : tmp + '-Movies'
+    const category = tmp !== 'movies' ? tmp : arr[arr.length - 2]
+    const { isLoading, data, isError, error } = useMoviesData()
```

Comment:
instead of useMoviesData which returns all the movies, use `useMoviewData` hook.